src/components/Explore/GroupBySelector.tsx

```diff
@@ -45,37 +80,64 @@ export function GroupBySelector({ options, radioAttributes, value, onChange, sho
-        let checks = !!options.find((o) => o.value === op);
+  const filterValidOptions = useMemo(() => {
+    const isOptionValid = (option: string) => {
+      // Check if option exists in dropdown
+      let isValid = !!options.find((o) => o.value === option);
 
-        // remove radio options that are in the filters
-        if (filters.find((f) => f.key === op && (f.operator === '=' || f.operator === '!='))) {
-          return false;
-        }
+      // Remove options that are in the filters
+      if (filters.find((f) => f.key === option && (f.operator === '=' || f.operator === '!='))) {
+        return false;
+      }
 
-        // if filters (primary signal) has 'Full Traces' selected, then don't add rootName or rootServiceName to options
-        // as you would overwrite it in the query if it's selected
-        if (filters.find((f) => f.key === 'nestedSetParent')) {
-          checks = checks && op !== 'rootName' && op !== 'rootServiceName';
-        }
+      // If filters (primary signal) has 'Full Traces' selected, then don't add rootName or rootServiceName to options
+      if (filters.find((f) => f.key === 'nestedSetParent')) {
+        isValid = isValid && option !== 'rootName' && option !== 'rootServiceName';
+      }
 
-        // if rate or error rate metric is selected, then don't add status to options
-        // as you would overwrite it in the query if it's selected
-        if (metricValue === 'rate' || metricValue === 'errors') {
-          checks = checks && op !== 'status';
-        }
+      // If rate or error rate metric is selected, then don't add status to options
+      if (metricValue === 'rate' || metricValue === 'errors') {
+        isValid = isValid && option !== 'status';
+      }
+
+      // Filter out ignored attributes
+      isValid = isValid && !ignoredAttributes.includes(option);
+
+      return isValid;
+    };
 
-        return checks;
-      })
+    const allOptionValues = options.map(op => op.value?.toString() ?? '');
+    
+    const validRadioOptions = radioAttributes.filter(isOptionValid);
+    
+    // Get valid other options (not in radio attributes, but valid and have been clicked)
+    const validOtherOptions = allOptionValues
+      .filter(option => 
+        isOptionValid(option) && 
+        !radioAttributes.includes(option) && 
+        (clickCounts[option] || 0) > 0
+      );
+
+    const allOptions = [...validRadioOptions, ...validOtherOptions];
+
+    return {
+      allOptions,
+      isOptionValid
+    };
+  }, [options, radioAttributes, filters, metricValue, clickCounts]);
+
+  const radioOptions = useMemo(() => {
+    let radioOptionsWidth = 0;
+    const { allOptions } = filterValidOptions;
+    
+    return allOptions
       .map((attribute) => ({
         label: attribute.replace(SPAN_ATTR, '').replace(RESOURCE_ATTR, ''),
         text: attribute,
         value: attribute,
+        clickCount: clickCounts[attribute] || 0
```

Comment:
There might be two potential issues with counting total clicks per label

1) The value is absolute per stack, not scoped to the selected data source, but each data source may have different set of labels to be prioritized. 
2) Using total values will create high values after some time making it hard for new labels to become favorites

Reply:
1. 👍  
2. Like a button in the UI to reset? That may clatter UI 🤔  I'll leave it up to @alexbikfalvi and @nadinevehling. I think pinning/unpinning could work a bit better but we can go with what we have in here and get feedback 👍